src/goto-programs/json_goto_trace.cpp

```diff
@@ -97,8 +97,8 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
-          assert(step.full_lhs_value.is_not_nil());
-          full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
+          if(step.full_lhs_value.is_not_nil())
+            full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
```

Comment:
I think it may actually be better to remove the `if` and treat the case of `nil` in `json()` so that we create some json objects which makes the problem more detectable than not adding anything.